osquery/tables/networking/linux/routes.cpp

```diff
@@ -131,6 +132,18 @@ void genNetlinkRoutes(const struct nlmsghdr* netlink_msg, QueryData& results) {
     case RTA_PRIORITY:
       r["metric"] = INTEGER(*(int*)RTA_DATA(attr));
       break;
+    case RTA_METRICS:
+      struct rtattr* xattr = static_cast<struct rtattr*> RTA_DATA(attr);
+      uint32_t xattr_size = RTA_PAYLOAD(attr);
+      while (RTA_OK(xattr, xattr_size)) {
+        switch (xattr->rta_type) {
```

Comment:
if(xattr->rta_type == RTAX_HOPLIMIT)

Reply:
I choose to use a `switch` here to make the code similar to the outer loop (the logic is the same).
Also the netlink message contains other extended attributes that may be added here in the future if necessary.

Reply:
Linux:
  - List of the extended attributes are listed in include/uapi/linux/rtnetlink.h;
  - Implementation: the iproute project is the reference: https://github.com/shemminger/iproute2/blob/master/ip/iproute.c#L523

BSD & Darwin:
  - Fields documentation: https://www.freebsd.org/cgi/man.cgi?query=rtentry
  - Ref. implementation: ndp/ndp.c